Observer/RefundOrderInventoryObserver.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Ampersand\DisableStockReservation\Observer;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Sales\Api\Data\OrderInterface;
+use Ampersand\DisableStockReservation\Service\ExecuteSourceDeductionForItems;
+use Magento\Sales\Model\Order\Creditmemo;
+
+/**
+ * Class RefundOrderInventoryObserver
+ * @package Ampersand\DisableStockReservation\Observer
+ */
+class RefundOrderInventoryObserver implements ObserverInterface
+{
+    /**
+     * @var ExecuteSourceDeductionForItems
+     */
+    private $executeSourceDeductionForItems;
+
+    /**
+     * RefundOrderInventoryObserver constructor.
+     * @param ExecuteSourceDeductionForItems $executeSourceDeductionForItems
+     */
+    public function __construct(
+        ExecuteSourceDeductionForItems $executeSourceDeductionForItems
+    ) {
+        $this->executeSourceDeductionForItems = $executeSourceDeductionForItems;
+    }
+
+    /**
+     * Return creditmemo items qty to stock
+     *
+     * @param EventObserver $observer
+     * @return void
+     */
+    public function execute(EventObserver $observer)
+    {
+        /** @var Creditmemo $creditmemo */
+        $creditmemo = $observer->getEvent()->getCreditmemo();
```

Comment:
We should probably add some type checking here to make sure we actually have a credit memo, so something like:
```php
if (!$creditmemo instanceof Creditmemo) {
    return $this;
}
```
There's a number of places throughout the PR we can add something like this, especially in observer methods as we can never be 100% that the object we want is attached to the event